src/commands/hosting-channel-deploy.ts

```diff
@@ -127,20 +142,31 @@ export default new Command("hosting:channel:deploy [channelId]")
         })
       );
 
-      await deploy(["hosting"], options, { hostingChannel: channelId });
+      const results = (await deploy(["hosting"], options, {
+        hostingChannel: channelId,
+      })) as DeployResult;
 
       logger.info();
       await syncAuthState(projectId, sites);
       const deploys: { [key: string]: ChannelInfo } = {};
+      const siteDetails = results.hosting;
       sites.forEach((d) => {
-        deploys[d.target || d.site] = d;
+        const siteKey = d.target || d.site;
+        if (siteDetails && siteKey in siteDetails) {
+          d.version = siteDetails[siteKey]["version"];
+        }
```

Comment:
What is the case where `siteDetails` is not defined?

Reply:
Good point. Removed.